interface/modules/custom_modules/oe-module-faxsms/src/Controller/FaxDocumentService.php

```diff
@@ -0,0 +1,377 @@
+     * Get fax document details
+     *
+     * @param string $faxSid Fax SID
+     * @return array|null Fax details or null if not found
+     */
+    public function getFaxDocument(string $faxSid): ?array
+    {
+        $fax = sqlQuery(
+            "SELECT * FROM oe_faxsms_queue WHERE job_id = ? AND site_id = ?",
+            [$faxSid, $this->siteId]
+        );
+        
+        if (empty($fax)) {
+            return null;
+        }
+        
+        $details = json_decode($fax['details_json'] ?? '{}', true);
+        
+        return [
+            'id' => $fax['id'],
+            'job_id' => $fax['job_id'],
+            'from' => $fax['calling_number'] ?? $details['from'] ?? 'Unknown',
+            'to' => $fax['called_number'] ?? $details['to'] ?? 'Unknown',
+            'date' => $fax['date'],
+            'status' => $fax['status'] ?? $details['status'] ?? 'unknown',
+            'patient_id' => $fax['patient_id'],
+            'document_id' => $fax['document_id'],
+            'media_path' => $fax['media_path'],
+            'direction' => $fax['direction'] ?? 'inbound',
+            'details' => $details
+        ];
+    }
+
+    /**
+     * Delete fax document
+     *
+     * @param string $faxSid Fax SID
+     * @param bool $deleteFile Whether to delete the physical file
+     * @return bool Success status
+     */
+    public function deleteFaxDocument(string $faxSid, bool $deleteFile = false): bool
+    {
+        try {
+            $fax = $this->getFaxDocument($faxSid);
+            
+            if (empty($fax)) {
+                return false;
+            }
+            
+            // Mark as deleted in queue
+            sqlStatement(
+                "UPDATE oe_faxsms_queue SET deleted = 1 WHERE job_id = ? AND site_id = ?",
+                [$faxSid, $this->siteId]
+            );
+            
+            // Optionally delete physical file
+            if ($deleteFile && !empty($fax['media_path']) && file_exists($fax['media_path'])) {
+                unlink($fax['media_path']);
+            }
+            
+            error_log("FaxDocumentService: Deleted fax {$faxSid}");
+            return true;
+        } catch (Exception $e) {
+            error_log("FaxDocumentService: Error deleting fax: " . $e->getMessage());
+            return false;
+        }
+    }
+
+    /**
+     * Get file extension from MIME type
+     *
+     * @param string $mimeType
+     * @return string
+     */
+    private function getExtensionFromMimeType(string $mimeType): string
+    {
+        $map = [
+            'application/pdf' => 'pdf',
+            'image/tiff' => 'tif',
+            'image/tif' => 'tif',
+            'image/jpeg' => 'jpg',
+            'image/png' => 'png'
+        ];
+        
+        return $map[$mimeType] ?? 'pdf';
+    }
+
+    /**
+     * Format phone number for display
+     *
+     * @param string $phone
+     * @return string
+     */
+    private function formatPhoneDisplay(string $phone): string
```

Comment:
I don't know if there's a standard way to do this in openemr right now, but there ought to be, so that the presentation is consistent.